toc/rfc/rfc-draft-pcap-cf.md

```diff
@@ -0,0 +1,129 @@
+containers.
+
+## Proposal: Custom Packet Capturing Tool
+
+This RFC proposes implementing a reduced-functionality packet capturing tool
+written in Go using the [gopacket][gopacket] library. This custom tool would be
+injected into application containers via Diego, similar to how [diego-ssh][diego-ssh]
+currently provides SSH access.
+
+### Key Features
+
+The tool reduces the attack surface to a minimum by only exposing the options
+which are stricly necessary to capture packets, namely:
+* Specifying a network interface
+* Applying packet filters (pcap-filter format)
+* Setting snapshot length (snaplen)
+
+The tool would be injected via Diego using a mechanism similar to
+[diego-ssh][diego-ssh]. This ensures that the tool is available in all apps
+regardeless of the deployment method and it follows established patterns that
+are already known by maintainers.
+
+The tool has the necessary file capabilities assigned (`CAP_NET_RAW` and
+`CAP_NET_ADMIN`) to be able to capture packets. It remains in the namespaces of
+the application to ensure a vulnerability in the tool does not enable a
+container escape.
+
+The tool is only accessible if SSH is enabled for the application.
+
+## Other Options Considered
+
+#### Option 1: `sudo` Access
+
+Adding a platform switch to make the `vcap` user a sudoer was considered but
+discarded due to:
+* Significant security risks of providing full root access in the app container.
+* Potential for privilege escalation beyond intended packet capture use
+* Inconsistency with Cloud Foundry's security-first design principles
+
+#### Option 2: `setcap` on `tcpdump` Binary  
+
+Setting capabilities on the existing `tcpdump` binary was considered but
+discarded because:
+* `tcpdump` offers functionality to run arbitrary code in its security
+  context
+* Broader attack surface compared to a purpose-built tool
+
+## Implementation Considerations
+
+Based on the proposed solution the following sections detail out the changes
+which have to be made to the individual components of Cloud Foundry to implement
+it.
+
+### diego-release: Custom Packet Capturing Tool
+
+A new package will be added to diego-release which implements packet capturing
+in go through the [gopacket][gopacket] library. The resulting binary will be included in
+the various lifecycle archives that are added to the final app container and
+the necessary capabilities (`CAP_NET_RAW` and `CAP_NET_ADMIN`) will be assigned
+to the executable via file capabilities. This allows regular users to gain those
+capabilities when executing the binary.
+
+### CF CLI: New `pcap` command
+
+Similar to the `bosh pcap` command a `cf pcap` command will be added. Like its
+predecessor it will connect to the desired instances via SSH and execute the new
+packet capturing tool and stream back the captured packets via stdout. If there
+are multiple streams, the CLI will merge them and write them out to a single
+file in the pcap format.
```

Comment:
```suggestion
Similar to the `bosh pcap` command, a `cf pcap` command will be added. Like its
counterpart, it will connect to the desired instances via SSH, execute the new
packet capturing tool and stream back the captured packets via stdout and thus via SSH. 
If there are multiple streams, the CLI will merge them and write them out to a single
file in the pcap format.
```

"predecessor" makes it look like bosh pcap might be going away.